src/api/userInsightTypes.ts

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright 2023 Cortex Applications, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+export enum UserInsightType {
+  ACTION_ITEM = 'ACTION_ITEM',
+  FAILING_RULE = 'FAILING_RULE',
+  NO_ENTITIES = 'NO_ENTITIES',
+  NO_ENTITIES_OWNED = 'NO_ENTITIES_OWNED',
+  NO_FAILING_RULES = 'NO_FAILING_RULES',
+  NO_SCORECARDS = 'NO_SCORECARDS',
+}
+
+export interface BaseUserInsight {
+  type: UserInsightType;
+}
+
+export interface ActionItemUserInsight extends BaseUserInsight {
+  entityId: number;
+  initiativeId: number;
+  ruleIds: string[];
+  targetDate: string;
+}
+
+export interface FailingRuleUserInsight extends BaseUserInsight {
+  entityId: number;
+  nextLadderLevel?: string;
+  ruleIds: string[];
+  scorecardId: number;
+  // TODO (dev-homepage): do we want to show next % if you complete rule(s) for percentage based insight
```

Comment:
Do you intend to address this TODO before merging this PR?

Reply:
no, removed it. was copied from `brain-app`